src/components/forms/ClassicVlanForm.js

```diff
@@ -0,0 +1,85 @@
+import React, { Component } from "react";
+import { IcseNameInput } from "../Inputs";
+import { IcseFormGroup } from "../Utils";
+import { IcseSelect } from "../Dropdowns";
+import {
+  buildFormDefaultInputMethods,
+  buildFormFunctions,
+} from "../component-utils";
+import PropTypes from "prop-types";
+
+class ClassicVlanForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { ...this.props.data };
+    this.handleInputChange = this.handleInputChange.bind(this);
+    buildFormDefaultInputMethods(this);
+    buildFormFunctions(this);
+  }
+
+  /**
+   * handle input change
+   * @param {event} event event
+   */
+  handleInputChange(event) {
+    this.setState(this.eventTargetToNameAndValue(event));
+  }
+
+  render() {
+    return (
+      <>
+        <IcseFormGroup>
+          <IcseNameInput
+            id={this.state.name + "-name"}
+            value={this.state.name}
+            onChange={this.handleInputChange}
+            componentProps={this.props}
+            hideHelperText
+            invalid={this.props.invalidCallback(this.state, this.props)}
+            invalidText={this.props.invalidTextCallback(this.state, this.props)}
+          />
+          <IcseSelect
+            labelText={"VLAN Type"}
+            name={"type"}
+            formName={this.props.data.name + "-type"}
+            groups={this.props.vlanTypes}
```

Comment:
this can always be `PUBLIC` and `PRIVATE`

should be title case when rendered, all caps in state